src/actions/clusterActions.js

```diff
@@ -300,128 +300,177 @@ export function clusterLoadApps(clusterId) {
-                catalog: app.catalog,
-                name: app.chartName,
-                namespace: app.namespace,
-                version: app.version,
-              },
+      // Create the App.
+      await appsApi
+        .createClusterApp(clusterID, app.name, {
+          body: {
+            spec: {
+              catalog: app.catalog,
+              name: app.chartName,
+              namespace: app.namespace,
+              version: app.version,
             },
-          })
-          .catch(error => {
-            if (error.status === 409) {
-              new FlashMessage(
-                `An app called <code>${app.name}</code> already exists on cluster <code>${clusterID}</code>`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            } else if (error.status === 400) {
-              new FlashMessage(
-                `Your input appears to be invalid. Please make sure all fields are filled in correctly.`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            } else {
-              new FlashMessage(
-                `Something went wrong while trying to install your app. Please try again later or contact support: support@giantswarm.io`,
-                messageType.ERROR,
-                messageTTL.LONG
-              );
-            }
-            throw error;
-          });
-      })
-      .then(() => {
-        dispatch({
-          type: types.CLUSTER_INSTALL_APP_SUCCESS,
-          clusterID,
-          app,
+          },
+        })
+        .catch(error => {
+          showAppInstallationErrorFlashMessage(app, clusterID, error);
+          throw error;
         });
 
-        new FlashMessage(
-          `Your app <code>${app.name}</code> is being installed on <code>${clusterID}</code>`,
-          messageType.SUCCESS,
-          messageTTL.MEDIUM
-        );
-      })
-      .catch(error => {
-        dispatch({
-          type: types.CLUSTER_INSTALL_APP_ERROR,
-          clusterID,
-          app,
-          error,
-        });
+      dispatch({
+        type: types.CLUSTER_INSTALL_APP_SUCCESS,
+        clusterID,
+        app,
+      });
 
-        throw error;
+      new FlashMessage(
+        `Your app <code>${app.name}</code> is being installed on <code>${clusterID}</code>`,
+        messageType.SUCCESS,
+        messageTTL.MEDIUM
+      );
+    } catch (error) {
+      dispatch({
+        type: types.CLUSTER_INSTALL_APP_ERROR,
+        clusterID,
+        app,
+        error,
       });
+
+      throw error;
+    }
   };
 }
 
+/**
+ * createAppConfig takes an app and a clusterID and tries to create the config map
+ * of that app. The app object must have a valuesYAML field representing the config
+ * map to be created.
+ * @param {object} app
+ * @param {string} clusterID
+ */
+async function createAppConfig(app, clusterID) {
+  const appConfigsApi = new GiantSwarm.AppConfigsApi();
+
+  return await appConfigsApi
```

Comment:
This would be bad practice, also impossible to debug.

[Please read up on this](https://eslint.org/docs/rules/no-return-await)

Reply:
Ok, removed the return. thx for the link